internal/admin/console/api.go

```diff
@@ -0,0 +1,561 @@
+							Name:      filter.Name,
+							Namespace: filter.Namespace,
+						})
+					}
+				}
+
+				// Process EnvoyProxy for GatewayClass
+				if res.EnvoyProxyForGatewayClass != nil {
+					configDump.EnvoyProxies = append(configDump.EnvoyProxies, ResourceSummary{
+						Name:      res.EnvoyProxyForGatewayClass.Name,
+						Namespace: res.EnvoyProxyForGatewayClass.Namespace,
+					})
+				}
+
+				// Process EnvoyProxies for Gateways
+				for _, proxy := range res.EnvoyProxiesForGateways {
+					if proxy != nil {
+						configDump.EnvoyProxies = append(configDump.EnvoyProxies, ResourceSummary{
+							Name:      proxy.Name,
+							Namespace: proxy.Namespace,
+						})
+					}
+				}
+
+				for _, backend := range res.Backends {
+					if backend != nil {
+						configDump.Backends = append(configDump.Backends, ResourceSummary{
+							Name:      backend.Name,
+							Namespace: backend.Namespace,
+						})
+					}
+				}
+
+				for _, serviceImport := range res.ServiceImports {
+					if serviceImport != nil {
+						configDump.ServiceImports = append(configDump.ServiceImports, ResourceSummary{
+							Name:      serviceImport.Name,
+							Namespace: serviceImport.Namespace,
+						})
+					}
+				}
+			}
+		}
+	}
+
+	return configDump
+}
+
+// handleAPIConfigDumpAll handles requests for complete provider resources dump
+func (h *Handler) handleAPIConfigDumpAll(w http.ResponseWriter, r *http.Request) {
+	if r.Method != http.MethodGet {
+		http.Error(w, "Method not allowed", http.StatusMethodNotAllowed)
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json; charset=utf-8")
+
+	if h.providerResources == nil {
+		// Return empty structure if no provider resources
+		emptyResponse := map[string]interface{}{
+			"message":   "No provider resources available",
+			"timestamp": time.Now(),
+		}
+		if err := json.NewEncoder(w).Encode(emptyResponse); err != nil {
+			http.Error(w, "Failed to encode response", http.StatusInternalServerError)
+		}
+		return
+	}
+
+	// Get the actual resources using GetResources() method
+	resources := h.providerResources.GetResources()
+
+	// Create a structured response with the actual resource data
+	response := map[string]interface{}{
+		"resources":  resources,
+		"timestamp":  time.Now(),
+		"totalCount": len(resources),
+	}
+
+	if err := json.NewEncoder(w).Encode(response); err != nil {
+		http.Error(w, "Failed to encode provider resources", http.StatusInternalServerError)
+		return
+	}
+}
+
+// handleAPIMetrics handles requests for metrics using the Prometheus registry
+func (h *Handler) handleAPIMetrics(w http.ResponseWriter, r *http.Request) {
+	if r.Method != http.MethodGet {
+		http.Error(w, "Method not allowed", http.StatusMethodNotAllowed)
+		return
+	}
+
+	// Create a Prometheus handler using the controller-runtime registry
+	// This includes all Envoy Gateway control plane metrics
+	handler := createCombinedMetricsHandler()
```

Comment:
is this another metrics server in addition to the one created by the metrics runner ?